### PR TITLE
Restore editor screenshot.

### LINKED ISSUE #44811: Screenshot in master only takes one screenshot
**Godot version:**
https://github.com/godotengine/godot/commit/b24f9f2d8f060533f2dff16c73bf93c94f23fbdf

**OS/device including version:**
win10, vulkan, nvidia

**Issue description:**
Wanted to take a screenshot of the editor. The second screenshot fails.

**Steps to reproduce:**
1. Take the second screenshot
2. The result is the same as the first screenshot

**Minimal reproduction project:**
Any project.

### LINKED ISSUE #45898: Editor's root viewport serves stale images after the first access for ViewportTexture
**Godot version:**
4.0 #4c528e7

**OS/device including version:**
Windows 10

**Issue description:**
Attempting to capture an editor screenshot with an editor plugin (GDScript) doesn't produce the desired result. The first time you try to get the image from the root `SubViewport`'s `ViewportTexture`, it is returned as expected. However, all consecutive attempts return the same exact image, even though the code even waits for two frames (as is tradition) before grabbing the picture:

```gdscript
	var editor_tree = get_editor_interface().get_base_control().get_tree()
	await editor_tree.process_frame
	await editor_tree.process_frame
	
	var base_control = get_editor_interface().get_base_control()
	var main_viewport = (base_control.get_tree().get_root() as Viewport)
	var viewport_image = main_viewport.get_texture().get_data()
	
	var datetime = OS.get_unix_time()
	viewport_image.save_png("res://.output/godot-editor-" + str(datetime) + ".png")
```

Same does not happen with a user project though, and the image is captured correctly.

```gdscript
	var image = $SubViewportContainer/SubViewport.get_texture().get_data()
	var datetime = OS.get_unix_time()
	image.save_png("res://.output/output-" + str(datetime) + ".png")
```

**Steps to reproduce:**
See the code above and demo projects.

**Minimal reproduction project:**
Editor plugin (press F11 to capture):
[plugin-editor-screenshotter-4.0.zip](https://github.com/godotengine/godot/files/5966876/plugin-editor-screenshotter-4.0.zip)

User project (press the on screen button to capture):
[bug-viewport-screenshot.zip](https://github.com/godotengine/godot/files/5966890/bug-viewport-screenshot.zip)

### PR DESCRIPTION
**Pull request by fire**
* Fixes https://github.com/godotengine/godot/issues/45898
* Fixes https://github.com/godotengine/godot/issues/44811

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -2888,11 +2889,11 @@ void EditorNode::_screenshot(bool p_use_utc) {
 
 void EditorNode::_save_screenshot(NodePath p_path) {
 	Control *editor_main_control = EditorInterface::get_singleton()->get_editor_main_control();
-	ERR_FAIL_COND_MSG(!editor_main_control, "Cannot get editor main control.");
-	Viewport *viewport = editor_main_control->get_viewport();
-	ERR_FAIL_COND_MSG(!viewport, "Cannot get editor main control viewport.");
-	Ref<ViewportTexture> texture = viewport->get_texture();
```
Comment (fire) [inline editor/editor_node.cpp:2894]: Why doesn't get_texture work directly? Investigating.

Comment (fire) [reply editor/editor_node.cpp:2894]: Can't move it to the viewport texture.

**Comment by fire** [comment]
I saw it working, then it didn't work. So strange.

**Comment by Mickeon** [comment]
Does this still require salvaging? Taking a screenshot works flawlessly to me.

Indeed, https://github.com/godotengine/godot/issues/44811 was fixed by https://github.com/godotengine/godot/pull/54427.

**Pull request closed without merging**